doc/source/reference/release_2.0.rst

```diff
@@ -10,6 +10,69 @@ Please send comments and questions to the networkx-discuss mailing list:
 
 API changes
 -----------
+* [`<>`_]
+  With the release of NetworkX 2.0 we are moving towards an iterator reporting API.
+  We used to have two methods for the same property of the graph, one that returns a
+  list and one that returns an iterator. With 2.0 we have removed the one that returns
+  a list and renamed the iterator returning method as the original one. For example,
+  ``G.nodes()`` used to return a list and ``G.nodes_iter()`` used to return an iterator, now
+  ``G.nodes()`` returns an iterator and ``G.nodes_iter()`` is removed. These changes have
+  been done for all the methods in all graph classes, Di/Multi/Graphs.
```

Comment:
Routines that used to return a dict now return an iterator of (key,value) 2-tuples, so that dict(G.degree()) returns a dict keyed by node with degree as value.